test/e2e/velero_utils.go

```diff
@@ -393,3 +396,77 @@ func waitForVSphereUploadCompletion(ctx context.Context, timeout time.Duration,
 
 	return err
 }
+
+func getVeleroVersion(ctx context.Context, veleroCLI string) (string, error) {
+	cmd := exec.CommandContext(ctx, veleroCLI, "version", "--timeout", "60s")
+	fmt.Println("Get Version Command:" + cmd.String())
+	stdout, stderr, err := veleroexec.RunCommand(cmd)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get velero version, stdout=%s, stderr=%s", stdout, stderr)
+	}
+
+	output := strings.Replace(stdout, "\n", " ", -1)
+	fmt.Println("Version:" + output)
+	regCompiler := regexp.MustCompile(`(?i)client\s*:\s*version\s*:\s*(\S+).+server\s*:\s*version\s*:\s*(\S+)`)
+	versionMatches := regCompiler.FindStringSubmatch(output)
+	if len(versionMatches) < 3 {
+		return "", errors.New("Velero version command returned null version")
+	}
+	if versionMatches[1] != versionMatches[2] {
+		return "", errors.New("Velero server and client version are not matched")
+	}
+	return versionMatches[1], nil
+}
+
+func checkVeleroVersion(ctx context.Context, veleroCLI string, expectedVer string) error {
+	tag := expectedVer
+	tagInstalled, err := getVeleroVersion(ctx, veleroCLI)
+	if err != nil {
+		return errors.WithMessagef(err, "Failed to get Velero version")
+	}
+	if strings.Trim(tag, " ") != strings.Trim(tagInstalled, " ") {
+		return errors.New(fmt.Sprintf("Velero version %s is not as expected %s", tagInstalled, tag))
+	}
+	fmt.Printf("Velero version %s is as expected %s\n", tagInstalled, tag)
+	return nil
+}
+
+func installVeleroCLI(version string) (string, error) {
+	name := "velero-" + version + "-" + runtime.GOOS + "-" + runtime.GOARCH
+	postfix := ".tar.gz"
+	tarball := name + postfix
+	toTmpPath := "/tmp/"
```

Comment:
Let's update this after the 1.7 release, but rather than using a fixed path here, we should use the in built [io/ioutil `TempDir`](https://pkg.go.dev/io/ioutil#TempDir) function as that takes care of any platform specific behaviour and allows us to create a temp path that we can clean up later.

Reply:
I modified this part as your suggestion, it's a very good point, I think I can fix it before RC, thanks a lot. And there is a small issue left here that I did not delete temp dir, because if I do it, it's kind of complicate to  cover the scenarios by different combinations of inputs. How about fix it in the future?

Reply:
Sounds good!